### DISCUSSION TITLE
Get-PnPContentType Format-List

### DISCUSSION THREAD

**Discussion by ingepepe** [pnp/powershell#431] [Q&A] [answered]
Is there a way to **see all the properties** of the content type? Format-List seems to not work for this:
```
PS> Get-PnPContentType -Identity Document

Name                           Id                             Group                          Description                   
----                           --                             -----                          -----------                   
Document                       0x0101                         Document Content Types         Create a new document.        



PS> Get-PnPContentType -Identity Document | Format-List


Format-List : The collection has not been initialized. It has not been requested or the request has not been executed. It may 
need to be explicitly requested.
At line:1 char:41
+ Get-PnPContentType -Identity Document | Format-List
+                                         ~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Format-List], CollectionNotInitializedException
    + FullyQualifiedErrorId : Microsoft.SharePoint.Client.CollectionNotInitializedException,Microsoft.PowerShell.Commands.Format 
   ListCommand
```

**Comment by veronicageek** [comment]
Hi - This was not an issue with the cmdlet, so I transferred it as a discussion.
What are you after? The site content types where 'Document' is the parent?

**Comment by LeonArmston** [comment, marked as answer]
CSOM (underlying api behind the most of the PnP cmdlets) by design rather than returning every property back only returns a subset of properties for Content Type i.e. Name, id etc. This is different from the on premise SharePoint PowerShell world where all properties would be returned.

If you want to retrieve any further properties you need to know the name of the property (can be found with Get-Member) and then explicitly request it using Get-PnPProperty

Run the following command piping Get-PnPContentType into the Get-Member cmdlet which will then give you a list of properties and methods that are available.
```powershell
Get-PnPContentType | Get-Member
```
![image](https://user-images.githubusercontent.com/12968962/111027819-962c3e80-83ea-11eb-8de3-eccf85c8f05d.png)

Identify the property that you want to retrieve i.e. Fields. 

1. Create a variable i.e. $contentType and assign it to Get-PnPContentType -Identity Document. 
2. Then use Get-PnPProperty with the ClientObject $contentType and specificy in the Property parameter the name of the property you would like to retrieve. This will then make an additional call to request the Fields property and adds it the $contentType variable. 
3. Finally return the $contentType variable as $contentType.Fields and it will show the extra Fields property values.

```powershell
$contentType = Get-PnPContentType -Identity Document
Get-PnPProperty -ClientObject $contentType -Property Fields
$contentType.Fields
```
![image](https://user-images.githubusercontent.com/12968962/111028028-d04a1000-83eb-11eb-9804-3cc57160fb2f.png)
